### PR TITLE
azurerm_logic_app_workflow - support open_authentication_policy

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is to support new properties `openAuthenticationPolicies` and `identity`.

--- PASS: TestAccLogicAppWorkflow_empty (334.20s)
--- PASS: TestAccLogicAppWorkflow_parameters (335.22s)
--- PASS: TestAccLogicAppWorkflow_requiresImport (353.68s)
--- PASS: TestAccLogicAppWorkflow_accessControl (466.46s)
--- PASS: TestAccLogicAppWorkflow_tags (551.24s)
--- PASS: TestAccLogicAppWorkflow_integrationAccount (666.87s)
--- PASS: TestAccLogicAppWorkflow_identity (822.87s)
--- PASS: TestAccLogicAppWorkflow_integrationServiceEnvironment (14333.72s)

### CODE REVIEW

File: internal/services/logic/logic_app_workflow_resource.go [hunk 1]
```diff
@@ -161,6 +198,45 @@ func resourceLogicAppWorkflow() *pluginsdk.Resource {
 				},
 			},
 
+			"identity": {
```
Comment (tombuildsstuff) [inline internal/services/logic/logic_app_workflow_resource.go:201]: can we update this to use the common schema from [the `identity` package](https://github.com/hashicorp/terraform-provider-azurerm/tree/main/internal/identity) here?

Comment (neil-yechenwei) [reply internal/services/logic/logic_app_workflow_resource.go]: Updated

**Comment by neil-yechenwei** [comment]
@tombuildsstuff & @katbyte , thanks for your comment. I've updated code. Please take another look. Thanks in advance.

File: internal/identity/independent_system_assigned_user_assigned.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+var _ Identity = IndependentSystemAssignedUserAssigned{}
+
+type IndependentSystemAssignedUserAssigned struct{}
+
+func (s IndependentSystemAssignedUserAssigned) Expand(input []interface{}) (*ExpandedConfig, error) {
```
Comment (tombuildsstuff) [inline internal/identity/independent_system_assigned_user_assigned.go:16]: @neil-yechenwei every instance I've seen of this is a Swagger bug - where the Service supports both System Assigned, User Assigned and System Assigned User Assigned but it's not correctly documented - can you confirm with the API/service team here?

Comment (neil-yechenwei) [reply internal/identity/independent_system_assigned_user_assigned.go:16]: Sure. Will do.

Comment (neil-yechenwei) [reply internal/identity/independent_system_assigned_user_assigned.go:16]: @tombuildsstuff , service team confirmed that they only support either SystemAssigned or UserAssigned or None for Logic Apps. They don’t support both at the same time. So, the swaggers are valid as of now.

**Comment by neil-yechenwei** [comment]
@tombuildsstuff, seems service team doesn't intend to implement "SystemAssigned,UserAssigned" for now.

**Comment by tombuildsstuff** [comment]
Dependent on https://github.com/hashicorp/go-azure-helpers/pull/92

**Comment by tombuildsstuff** [comment]
Dependent on https://github.com/hashicorp/go-azure-helpers/pull/93

File: internal/services/logic/logic_app_workflow_resource_test.go [hunk 3]
```diff
@@ -170,6 +170,42 @@ func TestAccLogicAppWorkflow_accessControl(t *testing.T) {
...
+	data := acceptance.BuildTestData(t, "azurerm_logic_app_workflow", "test")
+	r := LogicAppWorkflowResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.empty(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
```
Comment (tombuildsstuff) [inline internal/services/logic/logic_app_workflow_resource_test.go:183]: this is tested at the end, so by removing this we can check the Create path:

```suggestion
```

File: internal/services/logic/logic_app_workflow_resource.go [hunk 4]
```diff
@@ -800,6 +922,10 @@ func flattenLogicAppWorkflowAccessControlConfigurationPolicy(input *logic.FlowAc
 		result["allowed_caller_ip_address_range"] = flattenLogicAppWorkflowIPAddressRanges(input.AllowedCallerIPAddresses)
 	}
 
+	if input.OpenAuthenticationPolicies != nil && input.OpenAuthenticationPolicies.Policies != nil {
+		result["open_authentication_policy"] = flattenLogicAppWorkflowOpenAuthenticationPolicy(input.OpenAuthenticationPolicies.Policies)
+	}
```
Comment (tombuildsstuff) [inline internal/services/logic/logic_app_workflow_resource.go:927]: this should always have a value set into the state (as should this method)

**Review by tombuildsstuff — APPROVED** [review]
hey @neil-yechenwei 

Thanks for pushing those changes - I've taken a look through and left a few comments inline which I'm going to push a commit for, but this otherwise LGTM 👍 

Thanks!

File: internal/services/logic/logic_app_workflow_resource_test.go [hunk 5]
```diff
@@ -459,3 +523,64 @@ resource "azurerm_logic_app_workflow" "test" {
...
+  resource_group_name = azurerm_resource_group.test.name
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func (LogicAppWorkflowResource) updateIdentity(data acceptance.TestData) string {
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/logic/logic_app_workflow_resource_test.go]: Thanks.

File: internal/services/logic/logic_app_workflow_resource_test.go (hunk 3 above)
Comment (neil-yechenwei) [reply internal/services/logic/logic_app_workflow_resource_test.go]: Thanks.

File: internal/services/logic/logic_app_workflow_resource.go (hunk 4 above)
Comment (neil-yechenwei) [reply internal/services/logic/logic_app_workflow_resource.go]: Thanks for the update

**Comment by neil-yechenwei** [comment]
@tombuildsstuff , service team confirmed currently `open_authentication_policy` only supports `access_control.trigger`. It doesn't support `access_control.action`, `access_control.content` and `access_control.workflow_management`. So I only added this new property `open_authentication_policy` under `access_control.trigger`. So I added specific expand*/flatten* method to support it since you mentioned we have to set `open_authentication_policy` always.

**Review by katbyte — APPROVED** [review]
Thanks @neil-yechenwei - LGTM 💯

**Pull request merged**